tests/test_setup.py

```diff
@@ -175,6 +183,110 @@ def test_validate_platform_config(self):
             assert 'platform_conf' in self.hass.config.components
             assert not config['platform_conf']  # empty
 
+    def test_validate_platform_config_2(self):
+        """Test component PLATFORM_SCHEMA_BASE prio over PLATFORM_SCHEMA."""
+        platform_schema = PLATFORM_SCHEMA.extend({
+            'cheers': str,
+            'hello': str,
+        })
+        platform_schema_base = PLATFORM_SCHEMA.extend({
+            'hello': str,
```

Comment:
Eg if we do:
```suggestion
            'hello': 'world',
```
We can below check if platforms validate this item correctly, instead of adding other keys, since the schema should allow extra.